login_flows/santander.py

```diff
@@ -0,0 +1,123 @@
+# Script for https://netbank.santanderconsumer.dk/dk/login
+import requests, binascii, json, base64, argparse, sys, string, secrets, hashlib
+from bs4 import BeautifulSoup
+sys.path.append("..")
+from BrowserClient.BrowserClient import BrowserClient
+
+parser = argparse.ArgumentParser(description="argparser")
+parser.add_argument('--user', help='Your MitID username. For example: "GenericDanishCitizen"', required=True)
+parser.add_argument('--proxy', help='An optional socks5 proxy to use for all communication with MitID', required=False)
+parser.add_argument('--method', choices=['APP', 'TOKEN'], help='Which method to use when logging in to MitID, default APP', default='APP', required=False)
+args = parser.parse_args()
+
+user_id = args.user
+
+session = requests.Session()
+
+if args.proxy:
+    session.proxies.update({"http": f"socks5://{args.proxy}", "https": f"socks5://{args.proxy}" })
+    
+# First part of Santander procedure
+alphabet = string.ascii_letters + string.digits + '-._~'
+code_verifier = ''.join(secrets.choice(alphabet) for i in range(43))
+code_challenge = hashlib.sha256(code_verifier.encode('utf-8')).digest()
+code_challenge = base64.urlsafe_b64encode(code_challenge).decode('utf-8')
+code_challenge = code_challenge.replace('=', '')
```

Comment:
Replace the generation of "code_verifier", "code_challenge" and "state" with the functions I have created in login_flows/e-boks.py. And confirm that it still works after that. Hopefully these are always generated the same way :)

Reply:
Done